cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/upgrade/MetricsDataMigrator.java

```diff
@@ -134,13 +140,15 @@ public MetricsDataMigrator(final CConfiguration cConf, final DatasetFramework ds
     aggMetricStore = new DefaultMetricStore(factory, new int[]{Integer.MAX_VALUE});
   }
 
-  public void migrateMetricsTables(Version cdapVersion) {
-    if (cdapVersion.getVersion().equals("2.6")) {
-      migrateMetricsTableFromVersion26(cdapVersion);
+  public void migrateMetricsTables(ProjectInfo.Version cdapVersion) {
```

Comment:
this looks odd. You first convert 2.6_or_older or 2.7 into Version (in tool Main), then do other way around... Should you just pass that enum value from Main?

I was also thinking - does it make sense to run the tool I wrote before on the metrics table that you found in Main - to verify version? Then you'd reuse that enum